tutorials/01_introductory/lifecycle.py

```diff
@@ -0,0 +1,266 @@
+
+plt.rcParams.update({'figure.autolayout': True})
+
+fig, ax = plt.subplots()
+ax.bar(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+
+###############################################################################
+# Next, we'll add labels to the plot. To
+# do this with the OO interface, we can use the :meth:`axes.Axes.set` method
+# to set properties of this axis object.
+
+fig, ax = plt.subplots()
+ax.bar(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+ax.set(ylim=[0, 105], xlabel='Group ID', ylabel='Count',
+       title='Group ID and Count.')
+
+###############################################################################
+# We can also adjust the size of this plot using the :func:`pyplot.subplots`
+# function. We can do this with the ``figsize`` kwarg.
+#
+# .. note::
+#
+#    While indexing in python follows the form (row, column), the figsize
+#    kwarg follows the form (width, height). This follows conventions in
+#    visualization, which unfortunately are different from those of linear
+#    algebra.
+
+fig, ax = plt.subplots(figsize=(8, 4))
+ax.bar(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+ax.set(ylim=[0, 105], xlabel='Group ID', ylabel='Count',
+       title='Group ID and Count.')
+
+###############################################################################
+# For labels, we can specify custom formatting guidelines in the form of
+# functions by using the :class:`ticker.FuncFormatter` class. Below we'll
+# define a function that takes an integer as input, and returns a string
+# as an output.
+
+
+def make_labels(num, pos):
+    """The two args are the value and tick position."""
+    if num < 25:
+        s = 'super tiny'
+    elif num >= 25 and num < 50:
+        s = 'medium-ish'
+    elif num >= 50 and num < 75:
+        s = 'pretty big'
+    elif num > 75:
+        s = 'super big!'
+    else:
+        s = 'I dunno!'
+    return s
+
+formatter = FuncFormatter(make_labels)
+
+###############################################################################
+# We can then apply this formatter to the labels on our plot. To do this,
+# we'll use the ``xaxis`` attribute of our axis. This lets you perform
+# actions on a specific axis on our plot.
+
+fig, ax = plt.subplots(figsize=(8, 4))
+ax.bar(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+
+ax.set(ylim=[0, 105], xlabel='Group ID', ylabel='Count',
+       title='Group ID and Count.')
+ax.yaxis.set_major_formatter(formatter)
+
+###############################################################################
+# Combining multiple visualizations
+# =================================
+#
+# It is possible to draw multiple visualizations on the same instance of
```

Comment:
these are really multiple plot/ visualization elements...I think visualization is the term for the aggregate image